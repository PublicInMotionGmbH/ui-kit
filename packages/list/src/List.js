import React from 'react'
import PropTypes from 'prop-types'
import cls from 'classnames'

import { prefix } from '@talixo/commons'

const name = prefix('list')

/**
 * Component which represents List.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @returns {React.Element}
 */
function List (props) {
  const { bullet, children, className, style, ...passedProps } = props

  const mappedChildren = React.Children.map(children, child => {
    return (
      <li>
        <span>{bullet}</span>
        {child}
      </li>
    )
  })
  return (
    <ul className={cls(name, className)} {...passedProps} style={style}>
      {mappedChildren}
    </ul>
  )
}

List.propTypes = {
  /** Additional class name */
  className: PropTypes.string,

  /** Bullet type */
  bullet: PropTypes.node,

  /** Children */
  children: PropTypes.node,

  /** CSS styles */
  style: PropTypes.object
}

export default List
