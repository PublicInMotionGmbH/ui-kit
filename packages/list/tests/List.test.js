import React from 'react'
import List from '../src/List'
import { shallow } from 'enzyme'

describe('<List />', () => {
  it('renders children correctly', () => {
    const wrapper = shallow(
      <List bullet='#'>
        <span>Home</span>
        <span>Help</span>
        <span>Issues</span>
        <span>Log In</span>
      </List>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
