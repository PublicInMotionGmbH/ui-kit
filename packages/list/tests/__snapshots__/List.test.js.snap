// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders children correctly 1`] = `
<ul
  className="talixo-list"
>
  <li
    key=".0"
  >
    <span>
      #
    </span>
    <span>
      Home
    </span>
  </li>
  <li
    key=".1"
  >
    <span>
      #
    </span>
    <span>
      Help
    </span>
  </li>
  <li
    key=".2"
  >
    <span>
      #
    </span>
    <span>
      Issues
    </span>
  </li>
  <li
    key=".3"
  >
    <span>
      #
    </span>
    <span>
      Log In
    </span>
  </li>
</ul>
`;
